File: API/Entities/CourseProgram.cs
using API.Entities;

namespace API;

public class CourseProgram
{
    public Guid Id { get; set; }
    public int Orders { get; set; }
    public string Name { get; set; }
    public Guid CourseId { get; set; }
    public Course Course { get; set; }
    public List<ProgramDetail> ProgramDetails { get; set; }
}

File: client-app/src/app/types/courseProgram.ts
import { ProgramDetail } from "./programDetail";

export interface CourseProgram {
	id?: string;
	name: string;
	courseId: string;
    programDetails: ProgramDetail[]
}

export class CourseProgram implements CourseProgram {
	constructor(courseProgram?: CourseProgram) {
        if (courseProgram) {
            Object.assign(this, courseProgram);
        }
    }
}
